hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/upgrade/UpgradeDowngrade.java

```diff
@@ -275,4 +280,25 @@ protected Pair<Map<ConfigProperty, String>, List<ConfigProperty>> downgrade(Hood
       throw new HoodieUpgradeDowngradeException(fromVersion.versionCode(), toVersion.versionCode(), false);
     }
   }
+
+  /**
+   * Handles the case when auto-upgrade is disabled for upgrade operations.
+   * Sets compatible writer version and determines if upgrade logic should be skipped.
+   *
+   * @param fromVersion current table version
+   * @param toVersion target table version  
+   * @return true if upgrade logic should be skipped, false otherwise
+   */
+  private boolean handleAutoUpgradeDisabled(HoodieTableVersion fromVersion, HoodieTableVersion toVersion) {
+    if (!config.autoUpgrade() && fromVersion.versionCode() < toVersion.versionCode()) {
+      // Set appropriate writer version based on upgrade transition to maintain compatibility
+      if (fromVersion == HoodieTableVersion.SEVEN && toVersion == HoodieTableVersion.EIGHT) {
```

Comment:
`fromVersion` and `toVersion` can be two versions apart.  So should this consider more cases, not only the cases where `fromVersion` and `toVersion` are one version apart?

Reply:
I think you might be correct, so you are saying the case the fromVersion is six or 7 and toVersion is 9. We would need to account for those differences here as well, if autoUpgrade is disabled.